### PR TITLE
Add property to get GUID of the status message

### PR DESCRIPTION
**Pull request by ermshiperete**
This replaces PR #161.

(I thought we could use the `chorus-2.6` branch instead of maintaining a `lfmerge` branch, but I overlooked that the former uses libpalaso 3.1 which we can't use in lfmerge yet. Therefore I'm re-creating this PR agains the `lfmerge` branch)

This change requires the libpalaso changes from [#602](https://github.com/sillsdev/libpalaso/pull/602).

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/chorus/163)

### CODE REVIEW

File: src/LibChorus/merge/ChangeReport.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+namespace Chorus.merge
+{
+	public abstract class ChangeReport : IChangeReport
+	{
+		public FileInRevision ChildFileInRevision { get; private set; }
+		public FileInRevision ParentFileInRevision { get; private set; }
+		protected Guid _guid = GuidProvider.Current.NewGuid();
+
+		//        public override bool Equals(object obj)
```
Comment (jasonleenaylor) [inline src/LibChorus/merge/ChangeReport.cs]: I'm not a fan of committing commented out code, even though this was just merged. If there isn't an immediate plan for it we can just remove it.

Comment (ermshiperete) [reply src/LibChorus/merge/ChangeReport.cs]: Good point. Done.

**Comment by papeh** [comment]
Reviewed 3 of 4 files at r1, 8 of 11 files at r2, 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs, line 65 at r3](https://beta.reviewable.io/reviews/sillsdev/chorus/163#-L4wJ3vbCc8x5QOFp7Dh:-L4wJ3vbCc8x5QOFp7Di:ba1tmuw) ([raw file](https://github.com/sillsdev/chorus/blob/5269a0627260d496f7b56f5aab5b91853339c613/src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs#L65)):*
> ```Smalltalk
> 		{
> 			Assert.IsNotNull(_model.GetRepositoriesToList()[0].URI == "UsbKey");
> 			// Assert.IsNotNull(_model.GetRepositoriesToList().Any(r => r.URI == "UsbKey"));
> ```

commented code can be removed

Also, for either of these statements, I think you'd want to `Assert.True`

---

*[src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs, line 74 at r3](https://beta.reviewable.io/reviews/sillsdev/chorus/163#-L4wJlPoDKLGg07T9JoA:-L4wJlPoDKLGg07T9JoB:b-dsrnxh) ([raw file](https://github.com/sillsdev/chorus/blob/5269a0627260d496f7b56f5aab5b91853339c613/src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs#L74)):*
> ```Smalltalk
> 			_model = new SyncControlModel(_project, SyncUIFeatures.Advanced, null);
> 			_model.AddMessagesDisplay(_progress);
> 			Assert.AreEqual(1, _model.GetRepositoriesToList().Count);
> ```

should we also `Assert.That(_model.GetRepositoriesToList()[0].URI == "UsbKey")`?

---

*[src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs, line 120 at r3](https://beta.reviewable.io/reviews/sillsdev/chorus/163#-L4wKMFiATXf99SmI6MH:-L4wKMFiATXf99SmI6MI:b-ggux3t) ([raw file](https://github.com/sillsdev/chorus/blob/5269a0627260d496f7b56f5aab5b91853339c613/src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs#L120)):*
> ```Smalltalk
> 					Assert.Fail("Gave up waiting.");
> 				}
> 			}
> ```

could this boilerplate while loop be broken out into a method?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/sillsdev/chorus/163)*

**Comment by ermshiperete** [comment]
Review status: 11 of 14 files reviewed at latest revision, 4 unresolved discussions.

---

*[src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs, line 65 at r3](https://beta.reviewable.io/reviews/sillsdev/chorus/163#-L4wJ3vbCc8x5QOFp7Dh:-L4waVEX10V6lH-EO9rf:b-896fix) ([raw file](https://github.com/sillsdev/chorus/blob/5269a0627260d496f7b56f5aab5b91853339c613/src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs#L65)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

commented code can be removed

Also, for either of these statements, I think you'd want to `Assert.True`
</blockquote></details>

Done.

---

*[src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs, line 74 at r3](https://beta.reviewable.io/reviews/sillsdev/chorus/163#-L4wJlPoDKLGg07T9JoA:-L4waVvU9RKDBaEEG7e2:b-896fix) ([raw file](https://github.com/sillsdev/chorus/blob/5269a0627260d496f7b56f5aab5b91853339c613/src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs#L74)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

should we also `Assert.That(_model.GetRepositoriesToList()[0].URI == "UsbKey")`?
</blockquote></details>

Done.

---

*[src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs, line 120 at r3](https://beta.reviewable.io/reviews/sillsdev/chorus/163#-L4wKMFiATXf99SmI6MH:-L4waX0F5C908_xQhUeq:b-896fix) ([raw file](https://github.com/sillsdev/chorus/blob/5269a0627260d496f7b56f5aab5b91853339c613/src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs#L120)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

could this boilerplate while loop be broken out into a method?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/sillsdev/chorus/163)*

**Comment by papeh** [comment]
Reviewed 3 of 3 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs, line 32 at r4](https://reviewable.io:443/reviews/sillsdev/chorus/163#-L4x0rT6pSdT0iVOE91X:-L4x0rT6pSdT0iVOE91Y:b-pn7t2g) ([raw file](https://github.com/sillsdev/chorus/blob/93e269cfb5fc41a104ae909bc87fad80c1bcd49a/src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs#L32)):*
> ```Smalltalk
> 		}
> 
> 		private void WaitForSyncToFinish(SyncResults results)
> ```

I wonder if it would help to pass as reference (I don't think `results == null` will ever be false with the current code)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/chorus/163)*

**Comment by ermshiperete** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs, line 32 at r4](https://reviewable.io:443/reviews/sillsdev/chorus/163#-L4x0rT6pSdT0iVOE91X:-L4zMA3Z5h6-Qxjk_zoR:bd6sh14) ([raw file](https://github.com/sillsdev/chorus/blob/93e269cfb5fc41a104ae909bc87fad80c1bcd49a/src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs#L32)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

I wonder if it would help to pass as reference (I don't think `results == null` will ever be false with the current code)
</blockquote></details>

`SyncResults` is  a class, and before calling this method in the tests we set `results = null`, so it can and will evaluate to false.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/chorus/163)*

**Comment by papeh** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs, line 32 at r4](https://reviewable.io:443/reviews/sillsdev/chorus/163#-L4x0rT6pSdT0iVOE91X:-L59fBgCpI3THB8I5Oq-:b-2maev0) ([raw file](https://github.com/sillsdev/chorus/blob/93e269cfb5fc41a104ae909bc87fad80c1bcd49a/src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs#L32)):*
<details><summary><i>Previously, ermshiperete (Eberhard Beilharz) wrote…</i></summary><blockquote>

`SyncResults` is  a class, and before calling this method in the tests we set `results = null`, so it can and will evaluate to false.
</blockquote></details>

This works on my machine only when I pass `ref results`.  The value of `results` at call-time is `null`, but if we pass by reference, the while loop will know when the value of `results` is a valid pointer.

---

*[src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs, line 111 at r4](https://reviewable.io:443/reviews/sillsdev/chorus/163#-L59hec6enaBfmQh-922:-L59hec6enaBfmQh-923:b-pewvm) ([raw file](https://github.com/sillsdev/chorus/blob/93e269cfb5fc41a104ae909bc87fad80c1bcd49a/src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs#L111)):*
> ```Smalltalk
> 
> 		/// <summary>
> 		/// when r# allows categories, change this to just Catgory["SkipBehindProxy"]
> ```

now would be a good time to do this.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/chorus/163)*

**Comment by jasonleenaylor** [comment]
*[src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs, line 111 at r4](https://reviewable.io:443/reviews/sillsdev/chorus/163#-L59hec6enaBfmQh-922:-L5A-Ra36fGBVhxrRuZ1:b-68ur86) ([raw file](https://github.com/sillsdev/chorus/blob/93e269cfb5fc41a104ae909bc87fad80c1bcd49a/src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs#L111)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

now would be a good time to do this.
</blockquote></details>

It would also need to get one of the categories that we ignore on TC.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/chorus/163)*

**Comment by ermshiperete** [comment]
Review status: 13 of 14 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs, line 32 at r4](https://reviewable.io:443/reviews/sillsdev/chorus/163#-L4x0rT6pSdT0iVOE91X:-L5G797wNSNISmpWFp0-:b-ugc5jp) ([raw file](https://github.com/sillsdev/chorus/blob/93e269cfb5fc41a104ae909bc87fad80c1bcd49a/src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs#L32)):*
<details><summary><i>Previously, papeh wrote…</i></summary><blockquote>

This works on my machine only when I pass `ref results`.  The value of `results` at call-time is `null`, but if we pass by reference, the while loop will know when the value of `results` is a valid pointer.
</blockquote></details>

I don't understand this behaviour - I wrote a test app and that behaves as I'd expect, but the Jenkins Windows build matches your description.

I added the `ref`.

---

*[src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs, line 111 at r4](https://reviewable.io:443/reviews/sillsdev/chorus/163#-L59hec6enaBfmQh-922:-L5G7V48yj7n7DPaGLdU:bcj4pjl) ([raw file](https://github.com/sillsdev/chorus/blob/93e269cfb5fc41a104ae909bc87fad80c1bcd49a/src/Chorus.Tests/UI/Sync/SyncControlModelTests.cs#L111)):*
<details><summary><i>Previously, jasonleenaylor (Jason Naylor) wrote…</i></summary><blockquote>

It would also need to get one of the categories that we ignore on TC.
</blockquote></details>

Done. I'm not sure we need the "ignore on TC category" - want to try first without since the other tests work.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/chorus/163)*

**Comment by papeh** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/chorus/163#-:-L5Ji0VXbeYMmxb5wNJe:bnfp4nl)*

**Comment by ermshiperete** [comment]
> It would also need to get one of the categories that we ignore on TC.

Yep, @jasonleenaylor was right. Added the category in PR #165.

**Pull request merged**